src/scenic/core/object_types.py

```diff
@@ -3,666 +3,1760 @@
+        max_distances = [MeshVolumeRegion(mesh=shape.mesh, dimensions=max_bounds).inradius for shape in shapes]
+
+        distance_range = (min(min_distances), max(max_distances))
+
+        return InradiusHelper(support=distance_range)
+
+    @cached_property
+    def surface(self):
+        """A region containing the entire surface of this object"""
+        return self.occupiedSpace.getSurfaceRegion()
+
+    @cached_property
+    def onSurface(self):
+        """The surface used by the ``on`` specifier.
+
+        This region is used to sample position when
+        another object is placed ``on`` this object. By default
+        the top surface of this object (`topSurface`), but can
+        be overwritten by subclasses.
+        """
+        return self.topSurface
+
+    @cached_property
+    def rightSurface(self):
+        """A region containing the right surface of this object"""
+        return defaultSideSurface(self.occupiedSpace, dimension=0, \
+            positive=True, thresholds=self.sideComponentThresholds)
+
+    @cached_property
+    def leftSurface(self):
+        """A region containing the left surface of this object"""
+        return defaultSideSurface(self.occupiedSpace, dimension=0, \
+            positive=False, thresholds=self.sideComponentThresholds)
+
+    @cached_property
+    def frontSurface(self):
+        """A region containing the front surface of this object"""
+        return defaultSideSurface(self.occupiedSpace, dimension=1, \
+            positive=True, thresholds=self.sideComponentThresholds)
+
+    @cached_property
+    def backSurface(self):
+        """A region containing the back surface of this object"""
+        return defaultSideSurface(self.occupiedSpace, dimension=1, \
+            positive=False, thresholds=self.sideComponentThresholds)
+
+    @cached_property
+    def topSurface(self):
+        """A region containing the top surface of this object"""
+        return defaultSideSurface(self.occupiedSpace, dimension=2, \
+            positive=True, thresholds=self.sideComponentThresholds)
+
+    @cached_property
+    def bottomSurface(self):
+        """A region containing the bottom surface of this object"""
+        return defaultSideSurface(self.occupiedSpace, dimension=2, \
+            positive=False, thresholds=self.sideComponentThresholds)
+
+    def show_3d(self, viewer, highlight=False):
+        if needsSampling(self):
+            raise RuntimeError('tried to show() symbolic Object')
+
+        # Render the object
+        object_mesh = self.occupiedSpace.mesh.copy()
+
+        if highlight:
+            object_mesh.visual.face_colors = [30, 179, 0, 255]
+        elif self.color is not None:
+            object_mesh.visual.face_colors = self.color
+
+        viewer.add_geometry(object_mesh)
+
+        # If the camera is not a sphere, render the visible pyramid as a blue wireframe
+        if self.viewAngles != (math.tau, math.pi) or self.visibleDistance != 50:
```

Comment:
Why the condition on `visibleDistance`?

Reply:
This rule is intended to not clog up the viewer with every object's viewRegion, so it only shows the viewRegion if the Object has non default visibility parameters (since I figure that implies the user cares about visibility for it).

Reply:
Having a hard-coded number here is fragile, and in any case I don't think having non-default values is a good proxy for the user wanting to see the view region: someone might easily define a different default value in a subclass (as in fact several of the world models do). It used to make sense to show the ego's view region because of the built-in visibility requirement; now that that's gone, I don't think it does anymore. I'd suggest not rendering any of the view regions by default; we can have a `showVisibleRegion` property for controlling this for debugging purposes.

Reply:
Good point, done!